### ISSUE TITLE
suspend/suspend_advanced_auto failed to suspend but still passed with critical and high errors.

### ISSUE THREAD

**Issue by bladernr** [canonical/checkbox#2226]
### Bug Description

https://certification.canonical.com/hardware/202312-33278/submission/461841/
https://certification.canonical.com/hardware/202312-33278/submission/461841/test/892/result/54118161/

On that test run, the fwts suspend test failed yet the script marked it as passed.  The system never entered a suspend state.  This test should have failed given the three failures FWTS reported.

The test itself in the SRU suite is marked as a non-blocker, HOWEVER, the system failed to suspend and the test reported success anyway. Had validate_suspend not failed, we never would have known.

### Cert-blocker Test Case

- [x] cert-blocker

### To Reproduce

run the sru suite or retrigger the test run from https://test-observer.canonical.com/#/debs/278958 with 6.8.0-88.88 HWE

suspend should fail, and fwts should report a lot of stack traces and failure to suspend. but the scrpt actually passes.

269      set -o pipefail; checkbox-support-fwts_test -f none -l "$PLAINBOX_SESSION_SHARE"/suspend_single.log -s s3 --s3-sleep-delay=30 --s3-device-check --s3-device-check-delay=45 |      tee "$PLAINBOX_SESSION_SHARE"/suspend_single_times.log

### Expected Result

if suspend fails,  the job still reports success... this is because you use -f none with fwts_test.py which tells it to NEVER fail.

### Actual Result

suspend fails, and the job reports success.

### Environment

Jammy Desktop
HWE 6.8.0-88 (the suspend succeeded previously, this is new for the 88.88 kernel)
Release	22.04.5
Build stamp	canonical-oem-somerville-jammy-amd64-20220504-33+jellyfish-swellow+X111
Kernel	6.8.0-86-generic
Architecture	amd64
BIOS	Dell Inc.: 1.1.0 (UEFI)
Checkbox version	7.0.0~dev22~ubuntu22.04.1
Embedded controller firmware version	1.1
Memory	7.45GB total , 2048.0MB swap

### Relevant log output

```shell
Calling fwts
==================== Test Results ====================

1 iterations total.
Warning: Time to sleep was only reported 0 times out of 1 iterations!
Warning: Time to resume was only reported 0 times out of 1 iterations!

Average time to sleep: N/A
Average time to resume: N/A

Critical Failures: 1
 WARNING: The following test cases were reported as critical
 level failures by fwts:
  - sleep

 Please review the following log for more information:

================================= Iteration 1 ==================================
Results generated by fwts: Version V25.11.00 (2025-11-28 03:04:12).

Some of this work - Copyright (c) 1999 - 2025, Intel Corp. All rights reserved.
Some of this work - Copyright (c) 2010 - 2025, Canonical.
Some of this work - Copyright (c) 2016 - 2025, IBM.
Some of this work - Copyright (c) 2017 - 2025, ARM Ltd.

This test run on 05/12/25 at 03:24:06 on host Linux ubuntu 6.8.0-88-generic
#89~22.04.2-Ubuntu SMP PREEMPT_DYNAMIC Wed Oct 29 10:45:25 UTC 2 x86_64.

Command: "fwts -q --stdout-summary -r /var/tmp/checkbox-ng/sessions
/session_title-2025-12-04T19.10.37.session/session-share/suspend_single.log
--s3-sleep-delay 30 --s3-device-check --s3-device-check-delay 45 s3".
Running tests: s3.

s3: Sleep suspend/resume test.
--------------------------------------------------------------------------------
Test 1 of 1: Sleep suspend/resume test.
Defaulted to 1 test, use --s3-multiple=N to run more s2idle cycles 
s2idle cycle 1 of 1 
Detecting the power method.
Response to CanSuspend is yes 
User allowed to execute the CanSuspend action 
Using logind as the default power method.
Requesting Suspend action 
Skipping the minimum delay (0) and using a 3 seconds delay instead 
s2idle duration = 198.
wakeup source name: "ucsi-source-psy-USBC000:002" wakeup event was signaled.
wakeup source name: "00:01" wakeup event was signaled.
pm-action returned 0 after 198 seconds.
FAILED [CRITICAL] S2idleNotDeepest: Test 1, Expected /sys/kernel/debug/pmc_core
/slp_s0_residency_usec to increase from 308935964, got 308935964.
FAILED [HIGH] LongSuspend: Test 1, Unexpected: s2idle much longer than expected
(198 seconds).
FAILED [CRITICAL] PmDevicesFailedToSuspend: Test 1, CRITICAL Kernel message: [
1088.057357] PM: Some devices failed to suspend, or early wake event detected
Message repeated 1 times.

ADVICE: This is a bug picked up by the kernel, but as yet, the firmware test
suite has no diagnostic advice for this particular problem.

Found WARNING (1):
  WARNING: CPU: 15 PID: 22400 at /var/lib/dkms/nvidia/580.95.05/build/nvidia/nv.c:4455 nv_set_system_power_state+0x304/0x470 [nvidia]
  Modules linked in: ccm efi_test nfnetlink nvme_fabrics nvme_keyring tls rfcomm cmac algif_hash algif_skcipher af_alg bnep snd_soc_sof_sdw snd_soc_intel_hda_dsp_common snd_soc_intel_sof_maxim_common snd_sof_probes snd_soc_rt712_sdca_dmic snd_soc_rt1316_sdw snd_soc_rt712_sdca regmap_sdw_mbq regmap_sdw snd_soc_dmic snd_sof_pci_intel_mtl snd_sof_intel_hda_common soundwire_intel snd_sof_intel_hda_mlink soundwire_cadence snd_sof_intel_hda snd_hda_codec_hdmi snd_sof_pci snd_sof_xtensa_dsp snd_sof snd_sof_utils snd_soc_hdac_hda snd_hda_ext_core snd_soc_acpi_intel_match snd_soc_acpi soundwire_generic_allocation soundwire_bus snd_soc_core snd_compress ac97_bus snd_pcm_dmaengine snd_hda_intel snd_intel_dspcfg intel_uncore_frequency snd_intel_sdw_acpi intel_uncore_frequency_common dell_rbtn snd_hda_codec x86_pkg_temp_thermal intel_powerclamp snd_hda_core coretemp snd_hwdep dell_laptop snd_pcm btusb btrtl kvm_intel btintel btbcm btmtk snd_seq_midi ivsc_csi ivsc_ace snd_seq_midi_event binfmt_misc kvm snd_rawmidi
   irqbypass bluetooth snd_seq cmdlinepart hid_sensor_accel_3d dell_wmi hid_sensor_custom_intel_hinge snd_seq_device hid_sensor_gyro_3d hid_sensor_als spi_nor snd_timer hid_sensor_trigger dell_wmi_sysman mei_gsc_proxy dell_smbios joydev input_leds industrialio_triggered_buffer rapl ecdh_generic nls_iso8859_1 mei_vsc intel_rapl_msr dcdbas intel_cstate serio_raw dell_wmi_ddv dell_smm_hwmon firmware_attributes_class dell_wmi_descriptor ledtrig_audio kfifo_buf ecc mtd mei_me snd wmi_bmof hid_sensor_iio_common industrialio soundcore hid_multitouch mei dptf_pch_fivr iwlmvm mac80211 libarc4 intel_ipu6_isys(O) videobuf2_dma_contig videobuf2_memops videobuf2_v4l2 iwlwifi videobuf2_common processor_thermal_device_pci processor_thermal_device processor_thermal_wt_hint intel_ipu6_psys(O) processor_thermal_rfim processor_thermal_rapl cfg80211 intel_rapl_common intel_skl_int3472_tps68470 processor_thermal_wt_req intel_pmc_core tps68470_regulator processor_thermal_power_floor intel_ipu6(O) ov01a10(O) processor_thermal_mbox
   int3403_thermal clk_tps68470 v4l2_fwnode intel_vpu intel_vsec ipu_bridge igen6_edac int340x_thermal_zone mac_hid v4l2_async pmt_telemetry nvidia_uvm(POE) intel_hid int3400_thermal mei_vsc_hw acpi_thermal_rel intel_skl_int3472_discrete pmt_class acpi_tad sparse_keymap acpi_pad sch_fq_codel v4l2loopback(O) videodev mc msr parport_pc ppdev lp parport efi_pstore ip_tables x_tables autofs4 btrfs blake2b_generic raid10 raid456 async_raid6_recov async_memcpy async_pq async_xor async_tx xor raid6_pq libcrc32c raid1 raid0 dm_mirror dm_region_hash dm_log spi_ljca i2c_ljca gpio_ljca usbhid usb_ljca xe drm_gpuvm drm_exec gpu_sched drm_suballoc_helper drm_ttm_helper hid_sensor_custom r8153_ecm mmc_block cdc_ether hid_sensor_hub usbnet nvidia_drm(POE) nvidia_modeset(POE) r8152 intel_ishtp_hid mii nvidia(POE) i915 drm_buddy i2c_algo_bit crct10dif_pclmul crc32_pclmul polyval_clmulni ttm polyval_generic hid_generic ghash_clmulni_intel drm_display_helper rtsx_pci_sdmmc sha256_ssse3 psmouse sha1_ssse3 nvme intel_lpss_pci
   cec i2c_i801 spi_intel_pci i2c_smbus spi_intel rtsx_pci thunderbolt intel_lpss ucsi_acpi rc_core xhci_pci intel_ish_ipc nvme_core idma64 typec_ucsi xhci_pci_renesas intel_ishtp video nvme_auth typec i2c_hid_acpi i2c_hid hid wmi pinctrl_meteorlake aesni_intel crypto_simd cryptd
  CPU: 15 PID: 22400 Comm: nvidia-sleep.sh Tainted: P           OE      6.8.0-88-generic #89~22.04.2-Ubuntu
  Hardware name: Dell Inc. Precision 5490/, BIOS 1.2.1 03/18/2024
  RIP: 0010:nv_set_system_power_state+0x304/0x470 [nvidia]
  Code: ff ff 4c 89 f7 e8 2c 78 8a e7 e9 50 ff ff ff 0f 0b e9 34 fe ff ff 48 8b 3d 61 70 05 01 4c 89 fe e8 41 4c ae e6 e9 35 ff ff ff <0f> 0b 4c 89 f7 e8 02 78 8a e7 4d 85 ff 74 0d e8 48 cb d7 00 84 c0
  RSP: 0018:ffffcf390ee4ba48 EFLAGS: 00010202
  RAX: 0000000000000062 RBX: 0000000000000001 RCX: 0000000000000000
  RDX: 0000000000000000 RSI: 0000000000000000 RDI: ffffcf390ee4b978
  RBP: ffffcf390ee4ba80 R08: 0000000000000000 R09: 0000000000000000
  R10: 0000000000000000 R11: 0000000000000000 R12: 0000000000000062
  R13: ffff8dcc3aa40000 R14: ffff8dcc3aa40690 R15: ffff8dccfc2db000
  FS:  000070ba99186740(0000) GS:ffff8dcf7f780000(0000) knlGS:0000000000000000
  CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
  CR2: 000070ba9901b810 CR3: 0000000138d38005 CR4: 0000000000f70ef0
  DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
  DR3: 0000000000000000 DR6: 00000000ffff07f0 DR7: 0000000000000400
  PKRU: 55555554
  Call Trace:
   <TASK>
   nv_procfs_write_suspend+0x106/0x1c0 [nvidia]
   proc_reg_write+0x69/0xb0
   vfs_write+0xfd/0x480
   ? filp_flush+0x57/0x90
   ksys_write+0x73/0x100
   __x64_sys_write+0x19/0x30
   x64_sys_call+0x200f/0x2480
   do_syscall_64+0x81/0x170
   ? from_kgid_munged+0x12/0x30
   ? cp_new_stat+0x143/0x180
   ? __do_sys_newfstatat+0x53/0x90
   ? arch_exit_to_user_mode_prepare.constprop.0+0x1a/0xe0
   ? syscall_exit_to_user_mode+0x43/0x1e0
   ? do_syscall_64+0x8d/0x170
   ? handle_mm_fault+0xad/0x380
   ? arch_exit_to_user_mode_prepare.constprop.0+0x1a/0xe0
   ? irqentry_exit_to_user_mode+0x38/0x1e0
   ? irqentry_exit+0x43/0x50
   ? exc_page_fault+0x94/0x1b0
   entry_SYSCALL_64_after_hwframe+0x78/0x80
  RIP: 0033:0x70ba98f148f7
  Code: 10 00 f7 d8 64 89 02 48 c7 c0 ff ff ff ff eb b7 0f 1f 00 f3 0f 1e fa 64 8b 04 25 18 00 00 00 85 c0 75 10 b8 01 00 00 00 0f 05 <48> 3d 00 f0 ff ff 77 51 c3 48 83 ec 28 48 89 54 24 18 48 89 74 24
  RSP: 002b:00007ffc61fa3638 EFLAGS: 00000246 ORIG_RAX: 0000000000000001
  RAX: ffffffffffffffda RBX: 0000000000000008 RCX: 000070ba98f148f7
  RDX: 0000000000000008 RSI: 00005e4ce08dd3e0 RDI: 0000000000000001
  RBP: 00005e4ce08dd3e0 R08: 0000000000000000 R09: 00005e4ce08dd3e0
  R10: 000070ba9901acf0 R11: 0000000000000246 R12: 0000000000000008
  R13: 000070ba9901b780 R14: 000070ba99017600 R15: 000070ba99016a00
   </TASK>

Found WARNING (2):
  WARNING: CPU: 15 PID: 22400 at /var/lib/dkms/nvidia/580.95.05/build/nvidia/nv.c:4666 nv_set_system_power_state+0x320/0x470 [nvidia]
  Modules linked in: ccm efi_test nfnetlink nvme_fabrics nvme_keyring tls rfcomm cmac algif_hash algif_skcipher af_alg bnep snd_soc_sof_sdw snd_soc_intel_hda_dsp_common snd_soc_intel_sof_maxim_common snd_sof_probes snd_soc_rt712_sdca_dmic snd_soc_rt1316_sdw snd_soc_rt712_sdca regmap_sdw_mbq regmap_sdw snd_soc_dmic snd_sof_pci_intel_mtl snd_sof_intel_hda_common soundwire_intel snd_sof_intel_hda_mlink soundwire_cadence snd_sof_intel_hda snd_hda_codec_hdmi snd_sof_pci snd_sof_xtensa_dsp snd_sof snd_sof_utils snd_soc_hdac_hda snd_hda_ext_core snd_soc_acpi_intel_match snd_soc_acpi soundwire_generic_allocation soundwire_bus snd_soc_core snd_compress ac97_bus snd_pcm_dmaengine snd_hda_intel snd_intel_dspcfg intel_uncore_frequency snd_intel_sdw_acpi intel_uncore_frequency_common dell_rbtn snd_hda_codec x86_pkg_temp_thermal intel_powerclamp snd_hda_core coretemp snd_hwdep dell_laptop snd_pcm btusb btrtl kvm_intel btintel btbcm btmtk snd_seq_midi ivsc_csi ivsc_ace snd_seq_midi_event binfmt_misc kvm snd_rawmidi
   irqbypass bluetooth snd_seq cmdlinepart hid_sensor_accel_3d dell_wmi hid_sensor_custom_intel_hinge snd_seq_device hid_sensor_gyro_3d hid_sensor_als spi_nor snd_timer hid_sensor_trigger dell_wmi_sysman mei_gsc_proxy dell_smbios joydev input_leds industrialio_triggered_buffer rapl ecdh_generic nls_iso8859_1 mei_vsc intel_rapl_msr dcdbas intel_cstate serio_raw dell_wmi_ddv dell_smm_hwmon firmware_attributes_class dell_wmi_descriptor ledtrig_audio kfifo_buf ecc mtd mei_me snd wmi_bmof hid_sensor_iio_common industrialio soundcore hid_multitouch mei dptf_pch_fivr iwlmvm mac80211 libarc4 intel_ipu6_isys(O) videobuf2_dma_contig videobuf2_memops videobuf2_v4l2 iwlwifi videobuf2_common processor_thermal_device_pci processor_thermal_device processor_thermal_wt_hint intel_ipu6_psys(O) processor_thermal_rfim processor_thermal_rapl cfg80211 intel_rapl_common intel_skl_int3472_tps68470 processor_thermal_wt_req intel_pmc_core tps68470_regulator processor_thermal_power_floor intel_ipu6(O) ov01a10(O) processor_thermal_mbox
   int3403_thermal clk_tps68470 v4l2_fwnode intel_vpu intel_vsec ipu_bridge igen6_edac int340x_thermal_zone mac_hid v4l2_async pmt_telemetry nvidia_uvm(POE) intel_hid int3400_thermal mei_vsc_hw acpi_thermal_rel intel_skl_int3472_discrete pmt_class acpi_tad sparse_keymap acpi_pad sch_fq_codel v4l2loopback(O) videodev mc msr parport_pc ppdev lp parport efi_pstore ip_tables x_tables autofs4 btrfs blake2b_generic raid10 raid456 async_raid6_recov async_memcpy async_pq async_xor async_tx xor raid6_pq libcrc32c raid1 raid0 dm_mirror dm_region_hash dm_log spi_ljca i2c_ljca gpio_ljca usbhid usb_ljca xe drm_gpuvm drm_exec gpu_sched drm_suballoc_helper drm_ttm_helper hid_sensor_custom r8153_ecm mmc_block cdc_ether hid_sensor_hub usbnet nvidia_drm(POE) nvidia_modeset(POE) r8152 intel_ishtp_hid mii nvidia(POE) i915 drm_buddy i2c_algo_bit crct10dif_pclmul crc32_pclmul polyval_clmulni ttm polyval_generic hid_generic ghash_clmulni_intel drm_display_helper rtsx_pci_sdmmc sha256_ssse3 psmouse sha1_ssse3 nvme intel_lpss_pci
   cec i2c_i801 spi_intel_pci i2c_smbus spi_intel rtsx_pci thunderbolt intel_lpss ucsi_acpi rc_core xhci_pci intel_ish_ipc nvme_core idma64 typec_ucsi xhci_pci_renesas intel_ishtp video nvme_auth typec i2c_hid_acpi i2c_hid hid wmi pinctrl_meteorlake aesni_intel crypto_simd cryptd
  CPU: 15 PID: 22400 Comm: nvidia-sleep.sh Tainted: P        W  OE      6.8.0-88-generic #89~22.04.2-Ubuntu
  Hardware name: Dell Inc. Precision 5490/, BIOS 1.2.1 03/18/2024
  RIP: 0010:nv_set_system_power_state+0x320/0x470 [nvidia]
  Code: 01 4c 89 fe e8 41 4c ae e6 e9 35 ff ff ff 0f 0b 4c 89 f7 e8 02 78 8a e7 4d 85 ff 74 0d e8 48 cb d7 00 84 c0 0f 85 2c 01 00 00 <0f> 0b 48 c7 c7 f0 e3 bd c1 45 31 ed e8 df 77 8a e7 48 c7 c7 f0 e3
  RSP: 0018:ffffcf390ee4ba48 EFLAGS: 00010246
  RAX: 0000000000000000 RBX: 0000000000000001 RCX: 0000000000000000
  RDX: 0000000000000000 RSI: 0000000000000000 RDI: 0000000000000000
  RBP: ffffcf390ee4ba80 R08: 0000000000000000 R09: 0000000000000000
  R10: 0000000000000000 R11: 0000000000000000 R12: 0000000000000062
  R13: ffff8dcc3aa40000 R14: ffff8dcc3aa40690 R15: ffff8dccfc2db000
  FS:  000070ba99186740(0000) GS:ffff8dcf7f780000(0000) knlGS:0000000000000000
  CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
  CR2: 000070ba9901b810 CR3: 0000000138d38005 CR4: 0000000000f70ef0
  DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
  DR3: 0000000000000000 DR6: 00000000ffff07f0 DR7: 0000000000000400
  PKRU: 55555554
  Call Trace:
   <TASK>
   nv_procfs_write_suspend+0x106/0x1c0 [nvidia]
   proc_reg_write+0x69/0xb0
   vfs_write+0xfd/0x480
   ? filp_flush+0x57/0x90
   ksys_write+0x73/0x100
   __x64_sys_write+0x19/0x30
   x64_sys_call+0x200f/0x2480
   do_syscall_64+0x81/0x170
   ? from_kgid_munged+0x12/0x30
   ? cp_new_stat+0x143/0x180
   ? __do_sys_newfstatat+0x53/0x90
   ? arch_exit_to_user_mode_prepare.constprop.0+0x1a/0xe0
   ? syscall_exit_to_user_mode+0x43/0x1e0
   ? do_syscall_64+0x8d/0x170
   ? handle_mm_fault+0xad/0x380
   ? arch_exit_to_user_mode_prepare.constprop.0+0x1a/0xe0
   ? irqentry_exit_to_user_mode+0x38/0x1e0
   ? irqentry_exit+0x43/0x50
   ? exc_page_fault+0x94/0x1b0
   entry_SYSCALL_64_after_hwframe+0x78/0x80
  RIP: 0033:0x70ba98f148f7
  Code: 10 00 f7 d8 64 89 02 48 c7 c0 ff ff ff ff eb b7 0f 1f 00 f3 0f 1e fa 64 8b 04 25 18 00 00 00 85 c0 75 10 b8 01 00 00 00 0f 05 <48> 3d 00 f0 ff ff 77 51 c3 48 83 ec 28 48 89 54 24 18 48 89 74 24
  RSP: 002b:00007ffc61fa3638 EFLAGS: 00000246 ORIG_RAX: 0000000000000001
  RAX: ffffffffffffffda RBX: 0000000000000008 RCX: 000070ba98f148f7
  RDX: 0000000000000008 RSI: 00005e4ce08dd3e0 RDI: 0000000000000001
  RBP: 00005e4ce08dd3e0 R08: 0000000000000000 R09: 00005e4ce08dd3e0
  R10: 000070ba9901acf0 R11: 0000000000000246 R12: 0000000000000008
  R13: 000070ba9901b780 R14: 000070ba99017600 R15: 000070ba99016a00
   </TASK>

Found WARNING (3):
  WARNING: CPU: 5 PID: 22400 at /var/lib/dkms/nvidia/580.95.05/build/nvidia/nv.c:4405 nv_restore_user_channels+0x57/0x200 [nvidia]
  Modules linked in: ccm efi_test nfnetlink nvme_fabrics nvme_keyring tls rfcomm cmac algif_hash algif_skcipher af_alg bnep snd_soc_sof_sdw snd_soc_intel_hda_dsp_common snd_soc_intel_sof_maxim_common snd_sof_probes snd_soc_rt712_sdca_dmic snd_soc_rt1316_sdw snd_soc_rt712_sdca regmap_sdw_mbq regmap_sdw snd_soc_dmic snd_sof_pci_intel_mtl snd_sof_intel_hda_common soundwire_intel snd_sof_intel_hda_mlink soundwire_cadence snd_sof_intel_hda snd_hda_codec_hdmi snd_sof_pci snd_sof_xtensa_dsp snd_sof snd_sof_utils snd_soc_hdac_hda snd_hda_ext_core snd_soc_acpi_intel_match snd_soc_acpi soundwire_generic_allocation soundwire_bus snd_soc_core snd_compress ac97_bus snd_pcm_dmaengine snd_hda_intel snd_intel_dspcfg intel_uncore_frequency snd_intel_sdw_acpi intel_uncore_frequency_common dell_rbtn snd_hda_codec x86_pkg_temp_thermal intel_powerclamp snd_hda_core coretemp snd_hwdep dell_laptop snd_pcm btusb btrtl kvm_intel btintel btbcm btmtk snd_seq_midi ivsc_csi ivsc_ace snd_seq_midi_event binfmt_misc kvm snd_rawmidi
   irqbypass bluetooth snd_seq cmdlinepart hid_sensor_accel_3d dell_wmi hid_sensor_custom_intel_hinge snd_seq_device hid_sensor_gyro_3d hid_sensor_als spi_nor snd_timer hid_sensor_trigger dell_wmi_sysman mei_gsc_proxy dell_smbios joydev input_leds industrialio_triggered_buffer rapl ecdh_generic nls_iso8859_1 mei_vsc intel_rapl_msr dcdbas intel_cstate serio_raw dell_wmi_ddv dell_smm_hwmon firmware_attributes_class dell_wmi_descriptor ledtrig_audio kfifo_buf ecc mtd mei_me snd wmi_bmof hid_sensor_iio_common industrialio soundcore hid_multitouch mei dptf_pch_fivr iwlmvm mac80211 libarc4 intel_ipu6_isys(O) videobuf2_dma_contig videobuf2_memops videobuf2_v4l2 iwlwifi videobuf2_common processor_thermal_device_pci processor_thermal_device processor_thermal_wt_hint intel_ipu6_psys(O) processor_thermal_rfim processor_thermal_rapl cfg80211 intel_rapl_common intel_skl_int3472_tps68470 processor_thermal_wt_req intel_pmc_core tps68470_regulator processor_thermal_power_floor intel_ipu6(O) ov01a10(O) processor_thermal_mbox
   int3403_thermal clk_tps68470 v4l2_fwnode intel_vpu intel_vsec ipu_bridge igen6_edac int340x_thermal_zone mac_hid v4l2_async pmt_telemetry nvidia_uvm(POE) intel_hid int3400_thermal mei_vsc_hw acpi_thermal_rel intel_skl_int3472_discrete pmt_class acpi_tad sparse_keymap acpi_pad sch_fq_codel v4l2loopback(O) videodev mc msr parport_pc ppdev lp parport efi_pstore ip_tables x_tables autofs4 btrfs blake2b_generic raid10 raid456 async_raid6_recov async_memcpy async_pq async_xor async_tx xor raid6_pq libcrc32c raid1 raid0 dm_mirror dm_region_hash dm_log spi_ljca i2c_ljca gpio_ljca usbhid usb_ljca xe drm_gpuvm drm_exec gpu_sched drm_suballoc_helper drm_ttm_helper hid_sensor_custom r8153_ecm mmc_block cdc_ether hid_sensor_hub usbnet nvidia_drm(POE) nvidia_modeset(POE) r8152 intel_ishtp_hid mii nvidia(POE) i915 drm_buddy i2c_algo_bit crct10dif_pclmul crc32_pclmul polyval_clmulni ttm polyval_generic hid_generic ghash_clmulni_intel drm_display_helper rtsx_pci_sdmmc sha256_ssse3 psmouse sha1_ssse3 nvme intel_lpss_pci
   cec i2c_i801 spi_intel_pci i2c_smbus spi_intel rtsx_pci thunderbolt intel_lpss ucsi_acpi rc_core xhci_pci intel_ish_ipc nvme_core idma64 typec_ucsi xhci_pci_renesas intel_ishtp video nvme_auth typec i2c_hid_acpi i2c_hid hid wmi pinctrl_meteorlake aesni_intel crypto_simd cryptd
  CPU: 5 PID: 22400 Comm: nvidia-sleep.sh Tainted: P        W  OE      6.8.0-88-generic #89~22.04.2-Ubuntu
  Hardware name: Dell Inc. Precision 5490/, BIOS 1.2.1 03/18/2024
  RIP: 0010:nv_restore_user_channels+0x57/0x200 [nvidia]
  Code: 4c 89 f7 e8 7b bd 8a e7 f6 43 10 01 0f 84 80 00 00 00 48 89 de 31 ff e8 17 a1 d7 00 41 89 c7 85 c0 0f 84 6b 01 00 00 45 31 e4 <0f> 0b 49 81 c5 c8 07 00 00 4c 89 ef e8 48 bd 8a e7 be 01 00 00 00
  RSP: 0018:ffffcf390ee4ba10 EFLAGS: 00010202
  RAX: 0000000000000062 RBX: ffff8dcc3aa40000 RCX: 0000000000000000
  RDX: 0000000000000000 RSI: 0000000000000000 RDI: ffffcf390ee4b940
  RBP: ffffcf390ee4ba38 R08: 0000000000000000 R09: 0000000000000000
  R10: 0000000000000000 R11: 0000000000000000 R12: ffff8dcc2f790000
  R13: ffff8dcc3aa40000 R14: ffff8dcc3aa40690 R15: 0000000000000062
  FS:  000070ba99186740(0000) GS:ffff8dcf7f280000(0000) knlGS:0000000000000000
  CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
  CR2: 000078403c2ae000 CR3: 0000000138d38006 CR4: 0000000000f70ef0
  DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
  DR3: 0000000000000000 DR6: 00000000ffff07f0 DR7: 0000000000000400
  PKRU: 55555554
  Call Trace:
   <TASK>
   nv_set_system_power_state+0x353/0x470 [nvidia]
   nv_procfs_write_suspend+0x106/0x1c0 [nvidia]
   proc_reg_write+0x69/0xb0
   vfs_write+0xfd/0x480
   ? filp_flush+0x57/0x90
   ksys_write+0x73/0x100
   __x64_sys_write+0x19/0x30
   x64_sys_call+0x200f/0x2480
   do_syscall_64+0x81/0x170
   ? from_kgid_munged+0x12/0x30
   ? cp_new_stat+0x143/0x180
   ? __do_sys_newfstatat+0x53/0x90
   ? arch_exit_to_user_mode_prepare.constprop.0+0x1a/0xe0
   ? syscall_exit_to_user_mode+0x43/0x1e0
   ? do_syscall_64+0x8d/0x170
   ? handle_mm_fault+0xad/0x380
   ? arch_exit_to_user_mode_prepare.constprop.0+0x1a/0xe0
   ? irqentry_exit_to_user_mode+0x38/0x1e0
   ? irqentry_exit+0x43/0x50
   ? exc_page_fault+0x94/0x1b0
   entry_SYSCALL_64_after_hwframe+0x78/0x80
  RIP: 0033:0x70ba98f148f7
  Code: 10 00 f7 d8 64 89 02 48 c7 c0 ff ff ff ff eb b7 0f 1f 00 f3 0f 1e fa 64 8b 04 25 18 00 00 00 85 c0 75 10 b8 01 00 00 00 0f 05 <48> 3d 00 f0 ff ff 77 51 c3 48 83 ec 28 48 89 54 24 18 48 89 74 24
  RSP: 002b:00007ffc61fa3638 EFLAGS: 00000246 ORIG_RAX: 0000000000000001
  RAX: ffffffffffffffda RBX: 0000000000000008 RCX: 000070ba98f148f7
  RDX: 0000000000000008 RSI: 00005e4ce08dd3e0 RDI: 0000000000000001
  RBP: 00005e4ce08dd3e0 R08: 0000000000000000 R09: 00005e4ce08dd3e0
  R10: 000070ba9901acf0 R11: 0000000000000246 R12: 0000000000000008
  R13: 000070ba9901b780 R14: 000070ba99017600 R15: 000070ba99016a00
   </TASK>

Suspend/Resume Timings:
  Could not determine time to suspend.
  Could not determine time to resume.
Completed s2idle cycle(s) 
Found 1 errors in kernel log.
Found 1 PM related suspend issues.
PASSED: Test 1, No device errors detected.
PASSED: Test 1, No kernel oopses detected.
Found 3 kernel WARN_ON warnings.
Found 1 s2idle errors.
Found 2 errors and 0 oopses doing 1 suspend/resume cycle(s).
PASSED: Test 1, All suspends took less than 15.00 seconds.
PASSED: Test 1, All resumes took less than 15.00 seconds.

================================================================================
4 passed, 3 failed, 0 warning, 0 aborted, 0 skipped, 0 info only.
================================================================================


4 passed, 3 failed, 0 warning, 0 aborted, 0 skipped, 0 info only.

Test Failure Summary
================================================================================

Critical failures: 2
 s3: Expected /sys/kernel/debug/pmc_core/slp_s0_residency_usec to increase from 308935964, got 308935964.
 s3: CRITICAL Kernel message: [ 1088.057357] PM: Some devices failed to suspend, or early wake event detected

High failures: 1
 s3: Unexpected: s2idle much longer than expected (198 seconds).

Medium failures: NONE

Low failures: NONE

Other failures: NONE

Test           |Pass |Fail |Abort|Warn |Skip |Info |
---------------+-----+-----+-----+-----+-----+-----+
s3             |    4|    3|     |     |     |     |
---------------+-----+-----+-----+-----+-----+-----+
Total:         |    4|    3|    0|    0|    0|    0|
---------------+-----+-----+-----+-----+-----+-----+
```
